### PR TITLE
Use new request handlers framework from swarm

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**
(Requires swarm v4.6.0.)

### CODE REVIEW

**Comment by gavin-norman-sociomantic** [comment]
Updated with Mirror adaptation (the equivalent of https://github.com/sociomantic-tsunami/dhtproto/pull/93).

**Comment by gavin-norman-sociomantic** [comment]
Green! :tada:

**Comment by gavin-norman-sociomantic** [comment]
Updated to use the now released swarm v4.6.0. Unblocked.

File: src/dhtproto/node/neo/request/core/Mixins.d [hunk 1]
```diff
@@ -44,3 +44,38 @@ public template RequestCore ( )
...
+
+    ***************************************************************************/
+
+    public void initialise ( RequestOnConn connection, Object resources_object )
+    {
+        this.connection = connection;
+        this.resources = cast(IRequestResources)resources_object;
+        assert(this.resources !is null);
+    }
+}
```
Comment (nemanja-boric-sociomantic) [inline src/dhtproto/node/neo/request/core/Mixins.d:81]: Does this needs :police_car: ?

File: src/dhtproto/node/neo/request/Mirror.d [hunk 2]
```diff
@@ -354,40 +361,56 @@ public abstract scope class MirrorProtocol_v0
...
+
+    /***************************************************************************
+
+        Called by the connection handler after the supported code has been sent
+        back to the client.
 
+    ***************************************************************************/
+
+    private bool initialised_ok;
+    private bool start_suspended;
```
Comment (nemanja-boric-sociomantic) [inline src/dhtproto/node/neo/request/Mirror.d]: Order of the (undocumented) fields and the method docs doesn't match.

File: src/dhtproto/node/neo/request/core/Mixins.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/dhtproto/node/neo/request/core/Mixins.d:81]: Hmm, traditionally, we've only included release notes for user-facing (i.e. client) changes.

Comment (nemanja-boric-sociomantic) [reply src/dhtproto/node/neo/request/core/Mixins.d:81]: Yeah, I've just asked. I feel the same.

File: src/dhtproto/node/neo/request/Mirror.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/dhtproto/node/neo/request/Mirror.d]: Oh yeah, they should be documented.

Comment (gavin-norman-sociomantic) [reply src/dhtproto/node/neo/request/Mirror.d]: Updated.

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Pull request merged**